tests/graph_test.py

```diff
@@ -533,3 +543,48 @@ def test_setitem_raises_on_conflicting_input_nodes_in_ancestor() -> None:
     graph = cb.Graph(g1)
     with pytest.raises(ValueError, match="Node inputs differ for node 'b'"):
         graph['x'] = cb.Graph(g2)
+
+
+def test_getitem_keeps_only_relevant_indices() -> None:
+    g = nx.DiGraph()
+    g.add_edge('a', 'c')
+    g.add_edge('b', 'c')
+
+    graph = cb.Graph(g)
+    mapped = graph.map({'a': [1, 2, 3]})
+    assert mapped['a'].indices == {'dim_0': range(3)}
+    assert mapped['b'].indices == {}  # Not mapped
+    assert mapped['c'].indices == {'dim_0': range(3)}
+
+
+def test_setitem_with_mapped_operands_raises_on_conflict() -> None:
+    g = nx.DiGraph()
+    g.add_edge('a', 'c')
+    g.add_edge('b', 'c')
+
+    graph = cb.Graph(g)
+    mapped = graph.map({'a': [1, 2, 3]})
+    b = cb.Graph(nx.DiGraph()).map({'b': [11, 12]})
```

Comment:
Why did you make `b` a completely new graph? What happens when you map `graph` over both `'a'` and `'b'` independently and use `__setitem__`?

Reply:
In that case, the indices will not be in conflict, so that would be a different test?

In the current implementation I am not trying to detect compatible indices, so if naming clashes one would still get an error. I am adding a test showing/documenting this behavior.

Reply:
So this test is only about the index name not merging subgraphs that were mapped differently?

Reply:
On closer look, I think there is a loophole someone in the `__setitem__` logic around handling of mapped nodes. It is currently hidden by the rejection of compatible mappings, but might surface unnoticed if we implement that. Need to have a think.

Reply:
Rewrote the test, and preventing setting mapped nodes (for now), and added new tests also for that.